llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,231 @@
+import logging
+import uuid
+from typing import Any, Dict, List, Optional, Sequence, cast
+
+from llama_index.bridge.pydantic import BaseModel, Field, PrivateAttr  # type: ignore
+from llama_index.indices.service_context import ServiceContext
+from llama_index.schema import BaseNode, RelatedNodeInfo, TextNode
+from llama_index.vector_stores.types import (
+    BasePydanticVectorStore,
+    MetadataFilters,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+
+_logger = logging.getLogger(__name__)
+_import_err_msg = "`google.generativeai` package not found, please run `pip install google-generativeai`"
+_default_doc_id = "default-doc"
+
+
+google_service_context = ServiceContext.from_defaults(
+    # Avoids instantiating OpenAI as the default model.
+    llm=None,
+    # Avoids instantiating HuggingFace as the default model.
+    embed_model=None,
+)
+
+
+class GoogleVectorStore(BasePydanticVectorStore):
+    stores_text: bool = True
```

Comment:
nit, add comment here:

Semantic Retriever stores the document node's text as string and embeds the vectors on the server automatically.